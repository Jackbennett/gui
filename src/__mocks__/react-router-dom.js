import React from 'react';
import { compose, setDisplayName } from 'recompose';
import * as rrd from 'react-router-dom';
// Just render plain div with its children
// eslint-disable-next-line no-import-assign
rrd.BrowserRouter = compose(setDisplayName('BrowserRouter'))(({ children }) => <div>{children}</div>);

module.exports = rrd;
