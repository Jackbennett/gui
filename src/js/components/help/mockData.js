export const helpProps = {
  docsVersion: '',
  findLocalIpAddress: jest.fn,
  isHosted: false,
  isEnterprise: false,
  menderDebPackageVersion: 'master',
  menderVersion: 'master',
  menderArtifactVersion: 'master',
  org: {
    tenant_token: ''
  }
};
