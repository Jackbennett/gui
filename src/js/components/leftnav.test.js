import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import LeftNav from './leftnav';
import { defaultState, undefineds } from '../../../tests/mockData';

const mockStore = configureStore([thunk]);

describe('LeftNav Component', () => {
  let store;
  beforeEach(() => {
    store = mockStore({ ...defaultState });
  });

  it('renders correctly', async () => {
    const { baseElement } = render(
      <Provider store={store}>
        <MemoryRouter>
          <LeftNav />
        </MemoryRouter>
      </Provider>
    );
    const view = baseElement.firstChild.firstChild;
    expect(view).toMatchSnapshot();
    expect(view).toEqual(expect.not.stringMatching(undefineds));
  });
});
