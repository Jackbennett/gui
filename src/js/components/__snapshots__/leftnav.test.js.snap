// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeftNav Component renders correctly 1`] = `
<div>
  <ul
    class="MuiList-root MuiList-padding"
    style="padding: 0px;"
  >
    <a
      aria-current="page"
      class="MuiListItem-root navLink leftNav MuiListItem-gutters active"
      href="/"
      style="padding: 22px 16px 22px 42px;"
    >
      <div
        class="MuiListItemText-root"
        style="text-transform: uppercase;"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          Dashboard
        </span>
      </div>
    </a>
    <a
      class="MuiListItem-root navLink leftNav MuiListItem-gutters"
      href="/devices"
      style="padding: 22px 16px 22px 42px;"
    >
      <div
        class="MuiListItemText-root"
        style="text-transform: uppercase;"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          Devices
        </span>
      </div>
    </a>
    <a
      class="MuiListItem-root navLink leftNav MuiListItem-gutters"
      href="/releases"
      style="padding: 22px 16px 22px 42px;"
    >
      <div
        class="MuiListItemText-root"
        style="text-transform: uppercase;"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          Releases
        </span>
      </div>
    </a>
    <a
      class="MuiListItem-root navLink leftNav MuiListItem-gutters"
      href="/deployments"
      style="padding: 22px 16px 22px 42px;"
    >
      <div
        class="MuiListItemText-root"
        style="text-transform: uppercase;"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          Deployments
        </span>
      </div>
    </a>
  </ul>
  <ul
    class="MuiList-root MuiList-padding"
    style="padding: 0px; position: absolute; bottom: 30px; left: 0px; right: 0px;"
  >
    <a
      class="MuiListItem-root navLink leftNav MuiListItem-gutters"
      href="/help"
      style="padding: 16px 16px 16px 42px;"
    >
      <div
        class="MuiListItemText-root"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          Help
        </span>
      </div>
    </a>
    <li
      class="MuiListItem-root MuiListItem-gutters Mui-disabled"
      disabled=""
      style="padding: 16px 16px 16px 42px; color: rgb(148, 148, 149);"
    >
      <div
        class="MuiListItemText-root MuiListItemText-multiline"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          <div
            class="clickable"
          />
        </span>
        <p
          class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
        >
          <a
            href="https://docs.mender.io/development/release-information/open-source-licenses"
            rel="noopener noreferrer"
            style="font-size: 13px; position: relative; top: 6px; color: rgb(52, 122, 135);"
            target="_blank"
          >
            License information
          </a>
        </p>
      </div>
    </li>
  </ul>
</div>
`;
